pkg/operator/metrics/metrics.go

```diff
@@ -35,6 +42,12 @@ var (
 		Help: "Credentials requests with asserted conditions.",
 	}, []string{"condition"})
 
+	// Report on the mode CCO is operating under
+	metricCredentialsMode = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "cco_credentials_mode",
+		Help: "Track current mode the cloud-credentials-operator is functioning under.",
+	}, []string{"mode"})
```

Comment:
This seems a sufficient way to model it, if we get into more complicated scenarios in the future where more than one mode could be true, we can just report multiple 1's.